frontend/apps/web/app/(mgmt)/[account]/connections/[id]/permissions/page.tsx

```diff
@@ -0,0 +1,211 @@
+'use client';
+'use client';
+import { CloneConnectionButton } from '@/components/CloneConnectionButton';
+import ResourceId from '@/components/ResourceId';
+import Spinner from '@/components/Spinner';
+import { SubNav } from '@/components/SubNav';
+import OverviewContainer from '@/components/containers/OverviewContainer';
+import LearnMoreTag from '@/components/labels/LearnMoreTag';
+import { TestConnectionResult } from '@/components/permissions/Permissions';
+import PermissionsDataTable from '@/components/permissions/PermissionsDataTable';
+import { getPermissionColumns } from '@/components/permissions/columns';
+import { useAccount } from '@/components/providers/account-provider';
+import SkeletonForm from '@/components/skeleton/SkeletonForm';
+import { PageProps } from '@/components/types';
+import { Button } from '@/components/ui/button';
+import { toast, useToast } from '@/components/ui/use-toast';
+import { useGetConnection } from '@/libs/hooks/useGetConnection';
+import { useTestProgressConnection } from '@/libs/hooks/useTestPostgresConnection';
+import { getErrorMessage } from '@/util/util';
+import { PlainMessage } from '@bufbuild/protobuf';
+import {
+  CheckConnectionConfigResponse,
+  ConnectionRolePrivilege,
+  GetConnectionResponse,
+  PostgresConnectionConfig,
+} from '@neosync/sdk';
+import { UpdateIcon } from '@radix-ui/react-icons';
+import { ColumnDef } from '@tanstack/react-table';
+import Error from 'next/error';
+import { useEffect, useState } from 'react';
+import { KeyedMutator } from 'swr';
+import RemoveConnectionButton from '../components/RemoveConnectionButton';
+import { getConnectionComponentDetails } from '../components/connection-component';
+
+export default function PermissionsPage({ params }: PageProps) {
+  const id = params?.id ?? '';
+  const { account } = useAccount();
+  const { data, isLoading, mutate } = useGetConnection(account?.id ?? '', id);
+
+  const [pgConfig, setPgConfig] = useState<PostgresConnectionConfig | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (
+      !pgConfig &&
+      data?.connection?.connectionConfig?.config.case === 'pgConfig'
+    ) {
+      setPgConfig(data.connection.connectionConfig.config.value);
+    }
+  }, [data, pgConfig]); //TODO: should listen to the fields in these objects but we don't know which ones will be filled out
+
+  const {
+    data: validationRes,
+    isLoading: isLoadingValidation,
+    mutate: mutateValidation,
+  } = useTestProgressConnection(
+    account?.id ?? '',
+    pgConfig ?? new PostgresConnectionConfig({})
```

Comment:
You can just auto calculate the pg config based on the incoming data without the need for a useEffect and a separate state variable (simpler is better here).

```suggestion
    data?.connection?.connectionConfig?.config.case === 'pgConfig' ? data.connection.connectionConfig.config.value : new PostgresConnectionConfig({})
```

This allows you to completely remove the useState and useEffect to have the same outcome.